python_modules/dagster/dagster/core/asset_defs/assets_job.py

```diff
@@ -179,3 +194,39 @@ def _op():
         return io_manager.load_input(input_context_with_upstream)
 
     return _root_manager
+
+
+def build_partitions_info(
+    assets: Sequence[AssetsDefinition],
+) -> Tuple[PartitionsDefinition, Callable[[str], Mapping[str, Any]]]:
+    """
+    Determines whether the partitionings of the given assets satisfy the constraints of a regular
+    non-asset partitioned job. I.e., every asset has the same partitioning as every other asset,
+    and all the partition mappings are identity partition mappings.
+
+    If the partitionings do satisfy the constraints of a regular non-asset partitioned job, then
+    returns a PartitionsDefinition and tags_for_partition_fn that can be used to build a partitioned
+    job.
+    """
+    partitions_defs = {
+        assets_def.partitions_def for assets_def in assets if assets_def.partitions_def is not None
+    }
+
+    if len(partitions_defs) != 1:
+        return None, None
+
+    for assets_def in assets:
+        for parent_asset_key in assets_def.parent_asset_keys:
+            if not assets_def.get_partition_mapping(parent_asset_key).is_identity:
+                return None, None
```

Comment:
This is quite strict.  Seems like the norm would be that asset jobs are not partitioned.

I almost rather this doesn't exist than limit it in this way.  This adds a mapping concept to the computational partitions without completeness.

Reply:
This PR makes room for a somewhat more flexible approach: https://github.com/dagster-io/dagster/pull/5965.